cmd/dataset_tokenizer/dataset_tokenizer.go

```diff
@@ -502,17 +507,79 @@ func WriteContexts(outPath string, nextContext ContextsIterator,
 		}
 	}()
 
-	for {
-		context, more := <-contexts
-		if !more {
-			break
+	endpos := 0
+	var buf []byte
+	var contextSize int
+	var target int64
+
+	if sortSpec == "shuffle" {
+		fmt.Printf("Shuffling!\n")
+		for {
+			context, more := <-contexts
+			if !more {
+				break
+			}
+			binContext := context.ToBin()
+			if endpos == 0 {
+				contextSize = len(*binContext)
+				buf = make([]byte, contextSize)
+
```

Comment:
Run this through `gofmt`, or use Goland's auto format -- there's a hotkey to format a file.